### PR TITLE
[ActionSheet] Expose properties for the table

### LINKED ISSUE #4903: [ActionSheet] Expose properties to enable theming
Expose all properties to enable color, typography and shape theming.

### LINKED ISSUE #5039: [ActionSheet] Write test to confirm colors
Test should be written for:
1. The default color values
2. Setting colors for individually through the appropriate properties.
3. Applying a themer.

### PR DESCRIPTION
**Pull request by codeman7**
### Context
In order to add themers to MDCActionSheetController we need to expose all properties that will be themed.

### The problem
Currently there is no way clients can set the tint color, image rendering mode, or label color for MDCActionSheetController.

### The fix
This exposes those properties in order to allow them to be themed or set to custom values.

### Related conversations
#4984 

### Related issues
Closes #5039 
Closes #4903 

### Screenshots
| Before | After | Another After |
| ------ | ------ | ------ |
|simulator screen shot - iphone xs max - 2018-09-20 at 10 58 30|simulator screen shot - iphone xs max - 2018-09-20 at 10 59 34|simulator screen shot - iphone x - 2018-09-20 at 15 19 16|

### Videos
| Before | After | Another After |
|----- | ----- | ----- |
|before|after|after2|

### CODE REVIEW

File: components/ActionSheet/src/MDCActionSheetController.h [hunk 1]
```diff
@@ -159,6 +159,21 @@ __attribute__((objc_subclassing_restricted))
...
 
+/**
+ The color for the text for all action items within an action sheet.
+ */
+@property(nonatomic, strong, nullable) UIColor *actionTextColor;
+
+/**
+ The tint color for the action items within an action sheet.
+ */
+@property(nonatomic, strong, nullable) UIColor *tintColor;
```
Comment (jverkoey) [inline components/ActionSheet/src/MDCActionSheetController.h]: Consider `actionTintColor`?

File: components/ActionSheet/src/MDCActionSheetController.m [hunk 2]
```diff
@@ -386,4 +394,21 @@ - (void)updateFontsForDynamicType {
...
+#pragma mark - Table customization
+
+- (void)setActionTextColor:(UIColor *)actionTextColor {
+  _actionTextColor = actionTextColor;
+  [self.view setNeedsLayout];
+}
+
+- (void)setTintColor:(UIColor *)tintColor {
+  _tintColor = tintColor;
+  [self.view setNeedsLayout];
```
Comment (jverkoey) [inline components/ActionSheet/src/MDCActionSheetController.m]: This call to `setNeedsLayout` is surprising - can you share some more context? I typically wouldn't expect color changes to affect layout.

File: components/ActionSheet/tests/unit/MDCActionSheetTest.m [hunk 3]
```diff
@@ -18,6 +18,7 @@
 #import <XCTest/XCTest.h>
 
 #import "../../src/private/MDCActionSheetHeaderView.h"
+#import "../../src/private/MDCActionSheetItemTableViewCell.h"
```
Comment (jverkoey) [inline components/ActionSheet/tests/unit/MDCActionSheetTest.m]: Tip: consider adding a new test file for cells.

Comment (codeman7) [reply components/ActionSheet/tests/unit/MDCActionSheetTest.m]: Done

File: components/ActionSheet/src/MDCActionSheetController.m (hunk 2 above)
Comment (codeman7) [reply components/ActionSheet/src/MDCActionSheetController.m]: Done

File: components/ActionSheet/src/MDCActionSheetController.h (hunk 1 above)
Comment (codeman7) [reply components/ActionSheet/src/MDCActionSheetController.h]: Done

File: components/ActionSheet/src/MDCActionSheetController.m (hunk 2 above)
Comment (codeman7) [reply components/ActionSheet/src/MDCActionSheetController.m]: Removed setNeedsLayout but added reloadData. This is in the case that the action sheet is presented then a client changes the color.

File: components/ActionSheet/tests/unit/MDCActionSheetTableCellTest.m [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import "MDCActionSheetTestHelper.m"
```
[the remark this answers is not in the document]
Comment (ianegordon) [reply components/ActionSheet/tests/unit/MDCActionSheetTableCellTest.m]: Importing a .m?

**Review by ianegordon — DISMISSED** [review]

File: components/ActionSheet/tests/unit/MDCActionSheetTableCellTest.m (hunk 4 above)
[the remark this answers is not in the document]
Comment (codeman7) [reply components/ActionSheet/tests/unit/MDCActionSheetTableCellTest.m]: Added a header file

**Review by jverkoey — APPROVED** [review]

**Pull request merged**